src/actions/chatActions.js

```diff
@@ -0,0 +1,16 @@
+export const CHAT_ADD_MESSAGE = 'CHAT_ADD_MESSAGE';
+export const CHAT_SET_USER_STATUS = 'CHAT_SET_USER_STATUS';
+export const CHAT_CLEAR_MESSAGES = 'CHAT_CLEAR_MESSAGES';
+export const CHAT_SET_USERNAME = 'CHAT_SET_USERNAME';
+
+export const addMessage = ({ key, fromUsername, maybeSenderId, message }) =>
+    ({ type: CHAT_ADD_MESSAGE, key, fromUsername, maybeSenderId, message })
+
+export const clearMessages = () =>
+    ({ type: CHAT_CLEAR_MESSAGES })
+
+export const addUserStatus = ({ fromUsername, userStatus, created }) =>
+    ({ type: CHAT_SET_USER_STATUS, fromUsername, userStatus, created })
+
+export const setUsername = (username) =>
+    ({ type: CHAT_SET_USERNAME, username })
```

Comment:
This is a trade of when using these kind of shorthand arrow functions. If we just type:
```
export const setUsername = (username) =>
   { type: .... }
```

then it interpret it as a function body instead of an object and in that case just give us a syntax error.

Without the parenthesis around the returned object, it needs to look like this:
```
export const setUsername = (username) => {
   return { type: .... }
}
```

The shorthand arrow function let us omit the return statement but if we want to return an object we need to wrap it in parenthesis.